### ISSUE TITLE
sar skips long filesystem names

### ISSUE THREAD

**Issue by jaunix** [sysstat/sysstat#48]
If a filesystem is more than 72 characters in length, sar skips it.  

Curious how the 72 limit in rd_stats.h was arrived at, and whether it should be increased.  Or use MAX_FILE_LEN which is defined as 256 in common.h ?

    /* Maximum length of filesystem name */
    #define MAX_FS_LEN      72

Example:

    $ mount|grep lvname
    /dev/mapper/foo-longlvname12345678901234567890123456789012345678901234567890 on /mnt type ext2 (rw,relatime,seclabel)
    /dev/mapper/foo-shortlvname on /mnt2 type ext2 (rw,relatime,seclabel)
    $ ./sar -F 1 1
    Linux 3.18.7-100.fc20.x86_64 (localhost.localdomain)    03/13/2015      _x86_64_        (2 CPU)
    
    04:56:11 AM  MBfsfree  MBfsused   %fsused  %ufsused     Ifree     Iused    %Iused FILESYSTEM
    04:56:12 AM      6866     10850     61.24     66.45    902345    258647     22.28 /dev/sda5
    04:56:12 AM       170       113     39.88     46.61     76592       320      0.42 /dev/sda2
    04:56:12 AM        12         0      0.91      6.07      3061        11      0.36 /dev/mapper/foo-shortlvname
    
    Summary:     MBfsfree  MBfsused   %fsused  %ufsused     Ifree     Iused    %Iused FILESYSTEM
    Summary:         6866     10850     61.24     66.45    902345    258647     22.28 /dev/sda5
    Summary:          170       113     39.88     46.61     76592       320      0.42 /dev/sda2
    Summary:           12         0      0.91      6.07      3061        11      0.36 /dev/mapper/foo-shortlvname
    $

**Comment by jaunix** [comment]
The following minor change makes it work. Just chose an arbitrary size of 256 here, unsure what most appropriate value is though.

     $ git diff
     diff --git a/rd_stats.c b/rd_stats.c
     index faa33cc..2a0ef06 100644
     --- a/rd_stats.c
     +++ b/rd_stats.c
     @@ -2071,7 +2071,7 @@ void read_bus_usb_dev(struct stats_pwr_usb *st_pwr_usb, int nbr)
      void read_filesystem(struct stats_filesystem *st_filesystem, int nbr)
      {
             FILE *fp;
     -       char line[256], fs_name[MAX_FS_LEN], mountp[128];
     +       char line[256], fs_name[MAX_FS_LEN], mountp[MAX_FS_LEN];
             int fs = 0;
             struct stats_filesystem *st_filesystem_i;
             struct statvfs buf;
     @@ -2083,7 +2083,7 @@ void read_filesystem(struct stats_filesystem *st_filesystem, int nbr)
                     if (line[0] == '/') {
     
                             /* Read current filesystem name and mount point */
     -                       sscanf(line, "%71s %127s", fs_name, mountp);
     +                       sscanf(line, "%256s %256s", fs_name, mountp);
     
                             /* Replace octal codes */
                             oct2chr(mountp);
     diff --git a/rd_stats.h b/rd_stats.h
     index c8248df..03897fb 100644
     --- a/rd_stats.h
     +++ b/rd_stats.h
     @@ -26,7 +26,7 @@
      /* Maximum length of USB product string */
      #define MAX_PROD_LEN   48
      /* Maximum length of filesystem name */
     -#define MAX_FS_LEN     72
     +#define MAX_FS_LEN     256
     
      #define CNT_PART       1
      #define CNT_ALL_DEV    0
     $


     $ ./sadc -S XALL 1 3 /tmp/foo
     $ ./sar -F -f /tmp/foo
     Linux 3.18.7-100.fc20.x86_64 (localhost.localdomain)    03/14/2015      _x86_64_        (2 CPU)
     
     04:06:37 PM  MBfsfree  MBfsused   %fsused  %ufsused     Ifree     Iused    %Iused FILESYSTEM
     04:06:38 PM      6675     11041     62.32     67.53    896673    264319     22.77 /dev/sda5
     04:06:38 PM       170       113     39.88     46.61     76592       320      0.42 /dev/sda2
     04:06:38 PM        19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-longlvname12345678901234567890123456789012345678901234567890
     04:06:38 PM        19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-shortlvname
     04:06:39 PM      6675     11041     62.32     67.53    896673    264319     22.77 /dev/sda5
     04:06:39 PM       170       113     39.88     46.61     76592       320      0.42 /dev/sda2
     04:06:39 PM        19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-longlvname12345678901234567890123456789012345678901234567890
     04:06:39 PM        19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-shortlvname
     Summary:         6675     11041     62.32     67.53    896673    264319     22.77 /dev/sda5
     Summary:          170       113     39.88     46.61     76592       320      0.42 /dev/sda2
     Summary:           19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-longlvname12345678901234567890123456789012345678901234567890
     Summary:           19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-shortlvname
     $

**Comment by sysstat** [comment]
The MAX_FS_LEN value (which defaults to 72) was a kind of compromise, long enough for the majority of filesystems names (or so I hoped) but not too long as it is used in stats_filesystem structure definition:
```
struct stats_filesystem {
	unsigned long long f_blocks		__attribute__ ((aligned (16)));
	unsigned long long f_bfree		__attribute__ ((aligned (16)));
	unsigned long long f_bavail		__attribute__ ((aligned (16)));
	unsigned long long f_files		__attribute__ ((aligned (16)));
	unsigned long long f_ffree		__attribute__ ((aligned (16)));
	char 		   fs_name[MAX_FS_LEN]	__attribute__ ((aligned (16)));
};
```
The above structure is used to save **one** sample of statistics for **one** filesystem in the system activity daily data file. And my first idea was to prevent this file from taking too much space on disk. Maybe this is absurd nowadays with current HD capacities. Anyway do you really think that this value should be increased or is it only a possibility? If someone has an opinion about this, tell me...

**Comment by sysstat** [comment]
I'm suggesting here to increase MAX_FS_LEN to 128 instead of 72.
Doesn't look like a breakthrough but I still feel concerned about disk usage (and I think it should be enough for most users).
And BTW, it doesn't break compatibility with previous sa datafiles format.

**Comment by sysstat** [comment]
Fixed by commit a82d6ab.
